### Задание 1.
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.



In [3]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import pandas as pd

In [4]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [5]:
cov = ((zp - zp.mean()) * (ks - ks.mean())).sum() / (zp.shape[0] - 1) # Считаем не смещенную ковариацию

In [6]:
cov

10175.37777777778

In [7]:
np.cov(zp, ks, ddof=1)

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [8]:
cov / (zp.std(ddof=1) * ks.std(ddof=1))

0.8874900920739164

In [9]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [30]:
df=pd.DataFrame(data={'zp':zp, 'ks':ks})
df

zp   ks
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [31]:
df.corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

#### Ответ: Не смещенная Ковариация =  10175.37777777778; Корреляция Пирсона = 0.88749009

### Задание 2.
Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.


In [3]:
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [17]:
X=x.mean()
n = 10
std = x.std(ddof=1) # Берем несмещенную оценку т.к. считаем из выборки
alpha = 0.05
t = stats.t.ppf(alpha/2, df=n-1) # Используем T распределение, т.к. дисперсия генеральной совокупности не известна.


In [20]:
l_value = X+t*(std/(n**0.5))

In [21]:
r_value = X-t*(std/(n**0.5))

In [22]:
[l_value, r_value]

[110.55608365158724, 125.64391634841274]

#### Ответ: [110.55608365158724, 125.64391634841274]

### Задание 3.
Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.


In [28]:
X = 174.2
n = 27
std = 5
alpha = 0.05
z = stats.norm.ppf(alpha/2) # Используем стандартное нормальное распределение, т.к. известна дисперсия ген совокупности


In [29]:
l_value = X+z*(std/(n**0.5))

In [30]:
r_value = X-z*(std/(n**0.5))

In [31]:
[l_value, r_value]

[172.3140237765397, 176.08597622346028]

#### Ответ: [172.3140237765397, 176.08597622346028]